### PR TITLE
Adds DCOS os attribute to linux agents

### LINKED ISSUE #1330: DCOS agents cannot work in a mixed environment
**Is this a request for help?**: no

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): issue

---

**What version of acs-engine?**: vlabs

---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**

**What happened**:

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by yakman2020**
**What this PR does / why we need it**: Sets up the DCOS os attribute for linux agents (windows is handled elsewhere).  This allows  services and jobs to be dispatched to linux or windows agents in a mixed environment.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1330

**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1331)

### CODE REVIEW

**Comment by yakman2020** [comment]
go fmt'ed engine.go

File: pkg/acsengine/engine.go [hunk 1]
```diff
@@ -1008,13 +1008,27 @@ func getDCOSMasterCustomNodeLabels() string {
 
 func getDCOSAgentCustomNodeLabels(profile *api.AgentPoolProfile) string {
 	var buf bytes.Buffer
+    var attr_string string
 	buf.WriteString("")
+    if len(profile.OSType) > 0  {
```
Comment (jackfrancis) [inline pkg/acsengine/engine.go]: I think `if profile.OSType != ""` would be more idiomatic here.

File: pkg/acsengine/engine.go [hunk 2]
```diff
@@ -1008,13 +1008,27 @@ func getDCOSMasterCustomNodeLabels() string {
...
+    buf.WriteString(attr_string)
 	if len(profile.CustomNodeLabels) > 0 {
-		buf.WriteString("MESOS_ATTRIBUTES=")
 		for k, v := range profile.CustomNodeLabels {
-			buf.WriteString(fmt.Sprintf("%s:%s;", k, v))
+			buf.WriteString(fmt.Sprintf(";%s:%s", k, v))
 		}
 	}
+    attr_string = "\""
+    buf.WriteString(attr_string)
```
Comment (jackfrancis) [inline pkg/acsengine/engine.go]: Can we just combine these two lines into `buf.WriteString("\"")`? (or does a string literal fail to interpret the escaped double quote?)

File: pkg/acsengine/engine.go (hunk 1 above)
Comment (yakman2020) [inline pkg/acsengine/engine.go]: ./engine.go:1028: invalid operation: profile.Ports != "" (mismatched types []int and string)

See, osType is a string, but not a string

File: pkg/acsengine/engine.go (hunk 2 above)
Comment (yakman2020) [inline pkg/acsengine/engine.go]: done

**Pull request merged**